catroid/src/main/java/org/catrobat/catroid/content/CollisionScript.java

```diff
@@ -60,19 +60,22 @@ public String getSpriteToCollideWithName() {
 
 	public void setSpriteToCollideWithName(String spriteToCollideWithName) {
 		this.spriteToCollideWithName = spriteToCollideWithName;
-		updateSpriteToCollideWith();
+		updateSpriteToCollideWith(ProjectManager.getInstance().getCurrentlyEditedScene());
 	}
 
 	public Sprite getSpriteToCollideWith() {
-		updateSpriteToCollideWith();
+		updateSpriteToCollideWith(ProjectManager.getInstance().getCurrentlyEditedScene());
 		return spriteToCollideWith;
 	}
 
-	private void updateSpriteToCollideWith() {
-		if (spriteToCollideWithName != null
-				&& (spriteToCollideWith == null || !spriteToCollideWithName.equals(spriteToCollideWith.getName()))) {
-			Scene currentScene = ProjectManager.getInstance().getCurrentlyEditedScene();
-			spriteToCollideWith = currentScene.getSprite(spriteToCollideWithName);
+	public void updateSpriteToCollideWith(Scene scene) {
+		if (spriteToCollideWithName == null) {
+			spriteToCollideWith = null;
+		} else {
+			spriteToCollideWith = scene.getSprite(spriteToCollideWithName);
+			if (spriteToCollideWith == null) {
+				spriteToCollideWithName = null;
+			}
```

Comment:
Will be called when the project has been loaded or a sprite has been renamed/deleted.
If a sprite has been renamed/deleted `scene.getSprite(name)` will return null and the name will be nulled as well.

Reply:
Fixes following bug:
- Create a project containing two physics sprites.
- Go into the Scripts view of one of them. It should have a script like
```
When physical collision with
Other sprites name
```
- Change the collision sprite to `anything`
- Press back
- And go back into the Scripts view.

*What do you see?*
```
When physical collision with
Other sprites name
```

*What is the expected output?*
```
When physical collision with
anything
```

This is fixed now.
IMO there won't be any performance issues with my approach because it's called only for rendered bricks (and during loading) and the sprite list is (usually) very short. If really wanted, I sure can provide a more efficient way :wink:.
___
The inner if block is for deleted sprites. If a sprite is deleted, the `When physical collision with` changes to `anything`*

\* It changes to `anything` only if the project is loaded or the brick is rendered. So there are still problems, but this is the same behavior as in the last release.